src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,69 @@
+package duke.task;
+
+import duke.DukeException;
+
+/**
+ * The Task class represents a todo item with a description and can be marked / unmarked as complete.
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public Task(String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    protected void mark() {
```

Comment:
May I clarify why the function is named `mark()`? Perhaps `markDone()` could be considered, to better convey the intent that this function marks the task as done.